clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -619,22 +622,61 @@ public void beginTransaction() throws ProducerFencedException {
      * and should also not commit offsets manually (via {@link KafkaConsumer#commitSync(Map) sync} or
      * {@link KafkaConsumer#commitAsync(Map, OffsetCommitCallback) async} commits).
      *
-     * @throws IllegalStateException if no transactional.id has been configured or no transaction has been started
+     * @throws IllegalStateException if no transactional.id has been configured, no transaction has been started,
+     *                               or encounters unexpected group fencing exception which should only be returned from
```

Comment:
nit: I don't think there's any reason to mention this. Unexpected errors fall under `KafkaException`, which is listed below.